builder/chroot-script.sh

```diff
@@ -2,117 +2,30 @@
-      printf "%-10s %50s\n" lower: "${tolowercase}"
-    fi
-    printf "%-10s %50s\n" long: "${KEYID_long}"
-    printf "%-10s %50s\n" short: "${KEYID_short}"
-    echo ""
-  else
-    echo "usage: function {print|fpr|long|short} GPGKEY"
-  fi
-}
-
-
-function get_gpg(){
-  GPG_KEY="${1}"
-  KEY_URL="${2}"
-
-  clean_print print "${GPG_KEY}"
-  GPG_KEY=$(clean_print fpr "${GPG_KEY}")
-
-  if [[ "${KEY_URL}" =~ ^https?://* ]]; then
-    echo "loading key from url"
-    KEY_FILE=temp.gpg.key
-    wget -q -O "${KEY_FILE}" "${KEY_URL}"
-  elif [[ -z "${KEY_URL}" ]]; then
-    echo "no source given try to load from key server"
-#    gpg --keyserver "${KEYSERVER}" --recv-keys "${GPG_KEY}"
-    apt-key adv --keyserver "${KEYSERVER}" --recv-keys "${GPG_KEY}"
-    return $?
-  else
-    echo "keyfile given"
-    KEY_FILE="${KEY_URL}"
-  fi
-
-  FINGERPRINT_OF_FILE=$(gpg --with-fingerprint --with-colons "${KEY_FILE}" | grep fpr | rev |cut -d: -f2 | rev)
-
-  if [[ ${#GPG_KEY} -eq 16 ]]; then
-    echo "compare long keyid"
-    CHECK=$(clean_print long "${FINGERPRINT_OF_FILE}")
-  elif [[ ${#GPG_KEY} -eq 8 ]]; then
-    echo "compare short keyid"
-    CHECK=$(clean_print short "${FINGERPRINT_OF_FILE}")
-  else
-    echo "compare fingerprint"
-    CHECK=$(clean_print fpr "${FINGERPRINT_OF_FILE}")
-  fi
-
-  if [[ "${GPG_KEY}" == "${CHECK}" ]]; then
-    echo "key OK add to apt"
-    apt-key add "${KEY_FILE}"
-    rm -f "${KEY_FILE}"
-    return 0
-  else
-    echo "key invalid"
-    exit 1
-  fi
-}
-
-## examples:
-# clean_print {print|fpr|long|short} {GPGKEYID|FINGERPRINT}
-# get_gpg {GPGKEYID|FINGERPRINT} [URL|FILE]
-
 # device specific settings
 HYPRIOT_DEVICE="Raspberry Pi"
+RELEASE="stretch" # todo ideally this should be retrieved from a command like lsb_release, but it is installed later.
 
 # set up /etc/resolv.conf
 DEST=$(readlink -m /etc/resolv.conf)
 export DEST
 mkdir -p "$(dirname "${DEST}")"
 echo "nameserver 8.8.8.8" > "${DEST}"
 
-RELEASE="stretch" # todo ideally this should be retrieved from a command like lsb_release, but it is not installed.
+export DEBIAN_FRONTEND=noninteractive
+
+# Install dirmngr to allow downloading apt keys from keyserver
+apt-get update && apt-get install -y dirmngr
 
 # set up hypriot rpi repository for rpi specific kernel- and firmware-packages
-PACKAGECLOUD_FPR=418A7F2FB0E1E6E7EABF6FE8C2E73424D59097AB
-PACKAGECLOUD_KEY_URL=https://packagecloud.io/gpg.key
-get_gpg "${PACKAGECLOUD_FPR}" "${PACKAGECLOUD_KEY_URL}"
+apt-key adv --keyserver keyserver.ubuntu.com --recv-keys C2E73424D59097AB
 echo "deb https://packagecloud.io/Hypriot/rpi/debian/ ${RELEASE} main" > /etc/apt/sources.list.d/hypriot.list
 
 # set up Docker CE repository
-DOCKERREPO_FPR=9DC858229FC7DD38854AE2D88D81803C0EBFCD88
-DOCKERREPO_KEY_URL=https://download.docker.com/linux/raspbian/gpg
-get_gpg "${DOCKERREPO_FPR}" "${DOCKERREPO_KEY_URL}"
+apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 7EA0A9C3F273FCD8
```

Comment:
Has docker changed its key?

Reply:
No, it's the same one. I think one of them is the fingerprint for the public key, the other is for the private key Either way, they result in the same keys imported. See:
https://keyserver.ubuntu.com/pks/lookup?search=0x7EA0A9C3F273FCD8&op=vindex
https://keyserver.ubuntu.com/pks/lookup?search=0x8D81803C0EBFCD88&op=vindex